### PR TITLE
Application handler tests added

### PR DESCRIPTION
**Pull request by KateTheDev**

### CODE REVIEW

File: applications/ExecutionManager/test/application_handler_test.cpp [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    suoption.optionArg = "abc";
+    suoption.optionKind = ExecutionManager::StartupOptionKindEnum::commandLineLongForm;
+    suoption.optionName = "def";
+    pinfo.applicationName = " ";
+    pinfo.processName = " ";
+    pinfo.startOptions = {suoption};
+
+    EXPECT_CALL(*m_iosmock, fork()).WillOnce(Return(childProcessId));
+    EXPECT_CALL(*m_iosmock, execv(_,_)).WillOnce(Return(execvRes));
```
[the remark this answers is not in the document]
Comment (RomanLevkovych) [reply applications/ExecutionManager/test/application_handler_test.cpp]: @rBekesh we need your support: How can we check params passed to `execv` ?

File: applications/ExecutionManager/test/application_handler_test.cpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
+#include "application_handler.hpp"
+
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include <memory>
```
Comment (ulianaStashkiv) [inline applications/ExecutionManager/test/application_handler_test.cpp]: move before #include "gtest/gtest.h" & add empty line

File: applications/ExecutionManager/test/application_handler_test.cpp [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+using namespace std;
```
Comment (ulianaStashkiv) [inline applications/ExecutionManager/test/application_handler_test.cpp]: delete and use std::

File: applications/ExecutionManager/test/application_handler_test.cpp [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+{
+    const int expectedValue = 0;
+    const int childProcessId = 0;
+
+    ExecutionManager::ProcessInfo pinfo;
+
+    EXPECT_CALL(*m_iosmock, fork()).WillOnce(Return(childProcessId));
+    EXPECT_CALL(*m_iosmock, execv(_,_)).WillOnce(Return(expectedValue)); 
+
+	ExecutionManager::ApplicationHandler ah{std::move(m_iosmock)};
```
Comment (ulianaStashkiv) [inline applications/ExecutionManager/test/application_handler_test.cpp]: delete extra space

File: applications/ExecutionManager/test/application_handler_test.cpp [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+{
+    const int execvRes = 1;
+    const int childProcessId = 0;
+
+    ExecutionManager::ProcessInfo pinfo;
+
+    EXPECT_CALL(*m_iosmock, fork()).WillOnce(Return(childProcessId));
+    EXPECT_CALL(*m_iosmock, execv(_,_)).WillOnce(Return(execvRes)); 
+
+	ExecutionManager::ApplicationHandler ah{std::move(m_iosmock)};
```
Comment (ulianaStashkiv) [inline applications/ExecutionManager/test/application_handler_test.cpp]: delete extra space

File: applications/ExecutionManager/test/application_handler_test.cpp [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    ASSERT_THROW(ah.startProcess(pinfo), runtime_error);
+}
+
+TEST_F(ApplicationHandlerTest, ShouldSucceedToKillProcess)
+{
+    pid_t processId = 5;
+
+    EXPECT_CALL(*m_iosmock, kill(_, _));
+
+	ExecutionManager::ApplicationHandler ah{std::move(m_iosmock)};
```
Comment (ulianaStashkiv) [inline applications/ExecutionManager/test/application_handler_test.cpp]: delete extra space

**Comment by ulianaStashkiv** [comment]
Move all Mock to folder common mocks.hpp & mocks.cpp

File: applications/ExecutionManager/test/application_handler_test.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (rBekesh) [reply applications/ExecutionManager/test/application_handler_test.cpp]: You could try doing something like this:

    EXPECT_CALL(*m_iosmock, execv(_, _ ))
       .Times(1)
       .WillOnce(DoAll(Invoke([&](const char* procName,  char** listOfArguments)
                                  {
                                    std::string fullProcName = APPLICATIONS_PATH + pinfo.createRelativePath();
                                    
                                    ASSERT_STREQ(fullProcName.c_str(), procName);
                                    ASSERT_STREQ(pinfo.processName.c_str(), listOfArguments[0]);
                                    
                                    const std::string cmdOption = "--" + suoption.optionName + "=" + suoption.optionArg;

                                    ASSERT_STREQ(cmdOption.c_str(), listOfArguments[1]);
                                    ASSERT_EQ(nullptr, listOfArguments[2]);
                                  }),
                       Return(execvRes)));

File: applications/ExecutionManager/test/application_handler_test.cpp [hunk 7]
```diff
@@ -1,24 +1,16 @@
 #include "application_handler.hpp"
+#include "mocks.hpp"
```
Comment (ulianaStashkiv) [inline applications/ExecutionManager/test/application_handler_test.cpp]: use <> for include

File: applications/ExecutionManager/test/application_handler_test.cpp [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+        "abc", "def"},
+      {ExecutionManager::StartupOptionKindEnum::commandLineSimpleForm,
+        "abc", "def"}};
+  ExecutionManager::ProcessInfo pinfo =
+    {" ", " ", suoptions};
+
+    EXPECT_CALL(*m_iosmock, fork()).WillOnce(Return(childProcessId));
+    EXPECT_CALL(*m_iosmock, execv(_, _))
+      .Times(1)
+      .WillOnce(DoAll(Invoke([&](const char* procName,  char** listOfArguments)
```
Comment (ulianaStashkiv) [inline applications/ExecutionManager/test/application_handler_test.cpp]: Why do we need Times(1) & WillOnce() ?

File: applications/ExecutionManager/test/application_handler_test.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (rBekesh) [reply applications/ExecutionManager/test/application_handler_test.cpp]: Or you could define two ACTIONS for verification of arguments:
// For verification of process name, which is first argument in execv method
ACTION_P(CheckProcessName, fullProcName) 
{ 
  ASSERT_STREQ(fullProcName.c_str(), arg0);
}

// For verification of each argument in arguments list passed to execv as char**.
ACTION_P(CheckArg, argData) 
{ 
  ASSERT_STREQ(argData.second, arg1[argData.first]);
}

Second action though would require some additional std::pair variables to be created within test containing position in the array and value of each argument that needs to be verified. So final EXPECT_CALL would look like :

  std::string fullProcName = APPLICATIONS_PATH + pinfo.createRelativePath();
  const std::string cmdOption = "--" + suoption.optionName + "=" + suoption.optionArg;

  // Pairs containing argument info, first : position in arguments list, second : value.
  std::pair<int, const char*> procNameArg {0, pinfo.processName.c_str()};
  std::pair<int, const char*> suOptionArg {1, cmdOption.c_str()};
  std::pair<int, const char*> nullTerminatingArg {2, nullptr};

  EXPECT_CALL(*m_iosmock, execv(_, _))
    .WillOnce(DoAll(CheckProcessName(fullProcName),
                    CheckArg(procNameArg),
                    CheckArg(suOptionArg),
                    CheckArg(nullTerminatingArg),
                    Return(execvRes)));

**Pull request merged**